### PR TITLE
Settings improvements

### PR DESCRIPTION
**Pull request by boek**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/settings/RadioSearchEngineListPreference.kt [hunk 1]
```diff
@@ -29,24 +27,14 @@ class RadioSearchEngineListPreference : SearchEngineListPreference,
 
     override fun onBindViewHolder(holder: PreferenceViewHolder?) {
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/settings/RadioSearchEngineListPreference.kt]: An override is not still needed here.

File: app/src/main/java/org/mozilla/fenix/settings/SearchEngineListPreference.kt [hunk 2]
```diff
@@ -72,7 +76,14 @@ abstract class SearchEngineListPreference : Preference, CoroutineScope {
             return
         }
 
-        val defaultSearchEngine =
+        // To get the default search engine we have to pass in a name that doesn't exist
+        // https://github.com/mozilla-mobile/android-components/issues/3344
+        val defaultSearchEngine = context.components.search.searchEngineManager.getDefaultSearchEngine(
+            context,
+            "."
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/settings/SearchEngineListPreference.kt]: This should be a named constant value. Is a period something a user could enter as valid input?

Comment (boek) [reply app/src/main/java/org/mozilla/fenix/settings/SearchEngineListPreference.kt]: @colintheshots This is a hack (detailed in the comment). Wouldn't a constant hide it even more?

File: app/src/main/res/layout/search_engine_radio_button.xml [hunk 3]
```diff
@@ -2,19 +2,40 @@
...
 
-<RadioButton xmlns:android="http://schemas.android.com/apk/res/android"
-    android:id="@+id/radio_button"
-    android:layout_width="match_parent"
+
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
     android:layout_height="@dimen/search_engine_radio_button_height"
-    android:layout_gravity="start"
+    android:layout_width="match_parent"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
```
Comment (ekager) [inline app/src/main/res/layout/search_engine_radio_button.xml]: nit why is this in the middle?

**Pull request merged**